### ISSUE TITLE
Build works - but 'The fully qualified file name must be less than 260 characters' in VSBuild.ps1 v1.0.013

### ISSUE THREAD

**Issue by jochenvanwylick** [Microsoft/vso-agent-tasks#377]
Hi all,

I've set up continuous deployment for my vNext app using the VSO build agent. I've got it running, but I'm always getting this error in VSBuld.ps1. The build continues and the deployed app works fine - I just have this error that I would like to get rid of.

I suspect that the problem has to do with node or bower packages that that have created a deep tree-structure. It's not my project file path that's too long, I checked. 

````
2015-08-06T15:12:11.2975509Z Executing the powershell script: 
C:\_Apps\agent\tasks\VSBuild\1.0.13\VSBuild.ps1
 
2015-08-06T15:12:12.2037866Z ##[error]The specified path, file name, or both are too long. 
The fully qualified file name must be less than 260 characters, and the directory name must 
be less than 248 characters.
 
2015-08-06T15:12:12.2350397Z C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe 
"C:\-path with length 160 or so.xproj" /nologo /m /nr:false /fl 
/flp:"logfile=C:\-again path of length 150 max.log" 
/dl:CentralLogger,"C:\_Apps\agent\agent\worker\Microsoft.TeamFoundation.DistributedTask.MSBuild.Logger.dll"*ForwardingLogger,
"C:\_Apps\agent\agent\worker\Microsoft.TeamFoundation.DistributedTask.MSBuild.Logger.dll" /p:DeployOnBuild=true 
/p:WebPublishMethod=Package /p:PackageAsSingleFile=true /p:SkipInvalidConfigurations=true 
/p:PackageLocation=\"C:\_Apps\agent\_work\96521202\staging\" /p:platform="any cpu" 
/p:configuration="release" /p:VisualStudioVersion="14.0"
 
2015-08-06T15:12:12.5475633Z Build started 6-8-2015 17:12:12.
... and after a while ...
2015-08-06T15:12:48.3444957Z Build succeeded.
2015-08-06T15:12:48.3444957Z     0 Warning(s) 
2015-08-06T15:12:48.3444957Z     0 Error(s)
2015-08-06T15:12:48.3444957Z Time Elapsed 00:00:35.82
````

Can I help you guys debug this?

**Comment by chrisrpatterson** [comment]
I am quite sure this is due to the node packages.  We are steadily working on fixing the things we can but some of the underlying issues in the .NET framework are hitting us.  Some customers are working around that by deleting all of the node packages at the end of each build.

**Comment by jochenvanwylick** [comment]
OK, no worries. If there's anything I can do, let me know.

**Comment by shanavt** [comment]
Any update on this? I am facing the same issue. I checked my solution and do not see any files with long filepaths or names..Additional information on pointing the filenames that are violating the rules will be helpful.

**Comment by chrisrpatterson** [comment]
Do you have a node_modules directory or something else that has long paths?

**Comment by shanavt** [comment]
Yes.. I have Bower Components..

**Comment by brettpostin** [comment]
As a workaround, try using npm3. The node_modules folder structure is flat by default.

Since doing this a lot of our path headaches have gone away.

**Comment by tuespetre** [comment]
Here's what I had to do to resolve this issue:

- Update the VSBuild and MSBuild tasks using [tfx](https://github.com/Microsoft/tfs-cli)
- Update the agent from the TFS web interface
- Use npm 3.2.2 on the build server
- Delete existing workspaces from the build agent's _work folder

This takes care of that error when 'restore NuGet packages' is checked. Using a wildcard solution name is a different story though, because even with npm 3.2.2, some node_modules are still just too long. For example, bower 1.6.5 still has a deep dependency branch. So to bite the remaining issues, just be sure to explicitly specify the solution/project files' names on your MSBuild/VSBuild tasks.

**Comment by ericsciple** [comment]
MSBuild/VSBuild tasks have been fixed to handle long paths in the 1.95 agent. The deployment is rolling out over the course of the next couple weeks. Your agent will need to be updated to 1.95 to get the fix (will happen automatically when the agent is restarted, otherwise can perform via the admin web UI).
